src/generator/docs.ts

```diff
@@ -66,4 +71,7 @@ async function main() {
   });
   await Promise.all(promises);
 }
-main().catch(console.error);
+
+if (require.main === module) {
```

Comment:
this feels very Python(y) 😆 

_but, I like the approach for testing._

Reply:
How dare you 😆 There are a few motivations here.  Can you double check my math, and let me know if there's a better way?
- Each of these files are both individually executable, *and* importable as a library
- As you mentioned, testing is nice

Is there a better way to go about it?

Reply:
No, I think this is a good approach and will steal it.